src/EventStore.Core/TransactionLog/LogRecords/PrepareLogRecord.cs

```diff
@@ -42,6 +42,7 @@ public static class PrepareFlagsExtensions {
 
 	public class PrepareLogRecord : LogRecord, IEquatable<PrepareLogRecord>, IPrepareLogRecord<string> {
 		public const byte PrepareRecordVersion = 1;
+		private readonly ReadOnlyMemory<byte> _data;
```

Comment:
suggestion: move this line below `public ReadOnlyMemory<byte> Data { get; }`

Reply:
```suggestion
		public ReadOnlyMemory<byte> Data =>
			Flags.HasFlag(PrepareFlags.IsRedacted) ? NoData : _data;
			
		private readonly ReadOnlyMemory<byte> _data;
```

we can also make the `Data` property return `NoData` if it the event is redacted, otherwise return `_data`

event writes would still use `_data` irrespective of whether the event is redacted or not

Reply:
>we can also make the Data property return NoData if it the event is redacted, otherwise return _data
Implemented this in the constructors

Reply:
thanks @lakshdeepsingheventstore